src/client.js

```diff
@@ -4799,7 +4799,7 @@ MatrixClient.prototype._processRoomEventsSearch = function(searchResults, respon
     searchResults.highlights = Object.keys(highlights);
 
     // append the new results to our existing results
-    for (let i = 0; i < room_events.results.length; i++) {
+    for (let i = 0; i < (room_events.results ? room_events.results.length : 0); i++) {
```

Comment:
Could you extract this to a `resultsLength` or similar constant above the loop? Maybe personal style, but I think it would be nice to keep a ternary out of the loop header itself.

Reply:
Thanks for the suggestion. I implemented it now.